File: readh5.py
import pandas as pd


# filepath where your file is located 
filepath = '/Users/VENUS/Downloads/JIllian/Abby/average/peak_AUC_UnrewardedPort_DLS_z_score_DLS.h5'

# read file and make a dataframe
df = pd.read_hdf(filepath, key='df', mode='r')

# print dataframe
print(df)

File: readh5.m
% filepath to the file which needs to be read
filepath = '/Users/VENUS/Downloads/JIllian/habitEarly/average/peak_AUC_rwdNP_DMS_z_score_DMS.h5';
% access h5 file column names
info_data = h5read(filepath, '/df/axis0');
% access each column data in h5 file
data = h5read(filepath, '/df/block0_values');

% access h5 file row names
info_filename = h5read(filepath, '/df/axis1');

% construct a struct with three keys : filename, freq, amplitude
S.filename = info_filename;

% for reading freq and amplitude from z-score
S.freq = data(1,:)';
S.amplitude = data(2,:)';

% construct table from the above struct
T = struct2table(S);

% To access each column in table use "T.{column_name}"